src/deprecates.jl

```diff
@@ -43,3 +37,28 @@ findat(a::AbstractArray, b::AbstractArray) = findat!(Array{Int}(size(b)), a, b)
 
 @deprecate df(obj::StatisticalModel) dof(obj)
 @deprecate df_residual(obj::StatisticalModel) dof_residual(obj)
+
+@weights WeightVec
```

Comment:
I'm guessing using `depwarn` inside the `WeightVec` and `weights` methods is the correct approach here since we want to deprecate the functionality rather than the signature?